### DISCUSSION TITLE
[@types/node] cannot be type-checked with `isolatedModules: true`

### DISCUSSION THREAD

**Discussion by nicolo-ribaudo** [DefinitelyTyped/DefinitelyTyped#68285] [Issues with a @types package] [answered]
We are getting these errors in https://github.com/babel/babel/pull/16222 that are forcing us to use `skipLibCheck: true`:

```js
node_modules/@types/node/events.d.ts:878:14 - error TS1291: 'events' resolves to a type and must be marked type-only in this file before re-exporting when 'isolatedModules' is enabled. Consider using 'import type' where 'events' is imported.

878     export = events;
                 ~~~~~~

node_modules/@types/node/module.d.ts:314:14 - error TS1291: 'module' resolves to a type and must be marked type-only in this file before re-exporting when 'isolatedModules' is enabled. Consider using 'import type' where 'module' is imported.

314     export = module;
                 ~~~~~~

node_modules/@types/node/stream.d.ts:1700:14 - error TS1291: 'stream' resolves to a type and must be marked type-only in this file before re-exporting when 'isolatedModules' is enabled. Consider using 'import type' where 'stream' is imported.

1700     export = stream;
                  ~~~~~~


Found 3 errors.
```

We are using TS nightly (5.4.0-dev.20240120), but I'm still opening this discussion here because the error seems to be legit (under isolatedModules, you have to disambiguate when re-exporting imported values/types). TS5.4 will be promoted to "beta" tomorrow.

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
Should we add `// @ts-ignore` comment?

**Reply by nicolo-ribaudo** [reply]
The easiest solution would be to use `import type events = require("events")` instead of `import events = require("events")` so that then that `export = events` is not ambiguous, but I'm not sure about how this affects compatibility with old TS versions.

**Reply by Semigradsky** [reply]
> The easiest solution would be to use...

We can't, `events` is not a type only

**Reply by Semigradsky** [reply]
I see `'events' resolves to a type` in error message. But I don't know why it happens. Is it a bug in TS 5.4.0 or new behaviour?

**Reply by ghost** [reply]
Created PR #68291 with this change and it seems to break many dependent libs

**Reply by nicolo-ribaudo** [reply]
I'm opening an issue in the TS repo

**Reply by nicolo-ribaudo** [reply]
https://github.com/microsoft/TypeScript/issues/57136

**Comment by nicolo-ribaudo** [comment, marked as answer]
Fixed in TS: https://github.com/microsoft/TypeScript/pull/57148
